### DISCUSSION TITLE
Unique constraint failed on the constraint: `PersonalData_userId_key`

### DISCUSSION THREAD

**Discussion by ragokan** [prisma/prisma#14874] [Q&A]
The schema:
```prisma
model User {
  id           Int           @id @default(autoincrement())
  personalData PersonalData?
  // other fields
}

model PersonalData {
  id           Int           @id @default(autoincrement())
  user   User @relation(fields: [userId], references: [id], onDelete: Cascade, onUpdate: Cascade)
  userId Int  @unique
  // other fields
}
```

The operation: 
```ts
      await prisma.personalData.upsert({
        where: { userId },
        update: personalData,
        create: { ...personalData, userId },
      });
  ```
 
  When I do it, I get the error `Unique constraint failed on the constraint: PersonalData_userId_key`.
  
  Because I can't have this problem manually (when I do the operation manually, without using api), I can't totally test it and I don't know what causes the issue. Why do I get this warning even while doing the *upsert* operation.

**Comment by nurul3101** [comment]
Hi @ragokan 👋 

It seems the issue is with the existing data. I tested this with the models that you have provided and it seems to be working as expected.

schema.prisma
```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model User {
  id           Int           @id @default(autoincrement())
  personalData PersonalData?
  // other fields
}

model PersonalData {
  id     Int  @id @default(autoincrement())
  user   User @relation(fields: [userId], references: [id], onDelete: Cascade, onUpdate: Cascade)
  userId Int  @unique
  // other fields
}
```

index.ts
```ts
import { PrismaClient } from '@prisma/client';

const prisma = new PrismaClient({
  log: ['query'],
});

async function main() {
  const upsert = await prisma.personalData.upsert({
    where: { userId: 1 },
    update: {},
    create: { userId: 1 },
  });

  console.log(upsert);
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });

```

output:
```
> ts-node index.ts

prisma:query BEGIN
prisma:query SELECT "public"."PersonalData"."id" FROM "public"."PersonalData" WHERE "public"."PersonalData"."userId" = $1 OFFSET $2
prisma:query SELECT "public"."PersonalData"."id" FROM "public"."PersonalData" WHERE "public"."PersonalData"."userId" = $1
prisma:query SELECT "public"."PersonalData"."id", "public"."PersonalData"."userId" FROM "public"."PersonalData" WHERE "public"."PersonalData"."id" = $1 LIMIT $2 OFFSET $3
prisma:query COMMIT
{ id: 1, userId: 1 }
```

**Reply by ragokan** [reply]
Well, when I try it, I also don't get an error. But weirdly the route does. May you have any idea why?

**Comment by ragokan** [comment]
Update: I created a logger to see what is happening. Weirdly sometimes frontend send two post requests at the same time, so that it runs two upsert at the same time. Now, I am telling the front team to fix that. Thank you @nurul3101

**Reply by nurul3101** [reply]
Glad to hear that the root cause of issue is identified! 🙌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
